### PR TITLE
rpm 4.19.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/r/rpm.rb [hunk 1]
```diff
@@ -61,26 +65,29 @@ def install
...
+      -DCMAKE_INSTALL_FULL_LOCALSTATEDIR=#{var}
+      -DENABLE_NLS=ON
+      -DENABLE_PLUGINS=OFF
+      -DWITH_INTERNAL_OPENPGP=ON
+      -DWITH_OPENSSL=ON
+      -DRPM_VENDOR=#{tap.user.downcase}
+      -DENABLE_TESTSUITE=OFF
+    ]
+    if OS.mac?
+      args = args + %w[
```
Comment (SMillerDev) [inline Formula/r/rpm.rb:81]:
```suggestion
      args += %w[
```

**Comment by ZhongRuoyu** [comment]
I think we should consider making this Linux-only. It requires Linux-only `elfutils`, relies heavily on GNU extensions like `strchrnul` from `strings.h` and `GLOB_ONLYDIR` from `glob.h`, and has less (if any) practical usage outside of Linux.

**Comment by p-linnane** [comment]
This should definitely be Linux only.

**Review by ZhongRuoyu — APPROVED** [review]

File: Formula/r/rpmspectool.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ class Rpmspectool < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "3b27e1de9b1da031214a7cfd84e39491faca0f84fa616f1f358b8602c7961bca"
   end
 
+  depends_on :linux
```
Comment (chenrui333) [inline Formula/r/rpmspectool.rb:17]: :sad:

**Review by fxcoudert — APPROVED** [review]

File: Formula/r/rpm.rb [hunk 3]
```diff
@@ -24,67 +15,56 @@ class Rpm < Formula
...
-
-  uses_from_macos "bzip2"
-  uses_from_macos "zlib"
-
-  on_macos do
-    depends_on "libomp"
-  end
+  depends_on "zlib"
 
   conflicts_with "rpm2cpio", because: "both install `rpm2cpio` binaries"
```
Comment (ZhongRuoyu) [inline Formula/r/rpm.rb:41]: It seems that despite the `depends_on :linux` condition, the conflict still needs to  be `on_linux` only. `rpm2cpio`'s `conflicts_with` is scoped.

```
  rpm
    * Formula rpm2cpio should also have a conflict declared with rpm
```

**Review by ZhongRuoyu — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/r/rpm.rb [hunk 4]
```diff
@@ -30,61 +21,53 @@ class Rpm < Formula
...
+  depends_on "python@3.11" => [:build, :test]
+  depends_on "acl"
+  depends_on "bzip2"
+  depends_on "dbus"
+  depends_on "elfutils"
   depends_on "gettext"
   depends_on "libarchive"
+  depends_on "libcap"
   depends_on "libmagic"
+  depends_on :linux
```
[the remark this answers is not in the document]
Comment (kishaningithub) [reply Formula/r/rpm.rb:32]: Were you able to log that ticket @chenrui333 ?

File: Formula/r/rpmspectool.rb (hunk 2 above)
Comment (glensc) [reply Formula/r/rpmspectool.rb:15]: yeah. I used rpm to query packages information `rpm -qpil foo.rpm`

**Pull request merged**